Lib/multiprocessing/shared_memory.py

```
@@ -0,0 +1,573 @@
+
+    def close(self):
+        self.buf.release()
+        self._mmap.close()
+        self.close_fd()
+
+
+class SharedMemory:
+
+    def __new__(cls, *args, **kwargs):
+        if os.name == 'nt':
+            cls = WindowsNamedSharedMemory
+        else:
+            cls = PosixSharedMemory
+        return cls(*args, **kwargs)
+
+
+def shareable_wrap(
+    existing_obj=None,
+    shmem_name=None,
+    cls=None,
+    shape=(0,),
+    strides=None,
+    dtype=None,
+    format=None,
+    **kwargs
+):
+    augmented_kwargs = dict(kwargs)
+    extras = dict(shape=shape, strides=strides, dtype=dtype, format=format)
+    for key, value in extras.items():
+        if value is not None:
+            augmented_kwargs[key] = value
+
+    if existing_obj is not None:
+        existing_type = getattr(
+            existing_obj,
+            "_proxied_type",
+            type(existing_obj)
+        )
+
+        #agg = existing_obj.itemsize
+        #size = [ agg := i * agg for i in existing_obj.shape ][-1]
+        # TODO: replace use of reduce below with above 2 lines once available
+        size = reduce(
+            lambda x, y: x * y,
+            existing_obj.shape,
+            existing_obj.itemsize
+        )
+
+    else:
+        assert shmem_name is not None
+        existing_type = cls
+        size = 1
+
+    shm = SharedMemory(shmem_name, size=size)
+
+    class CustomShareableProxy(existing_type):
+
+        def __init__(self, *args, buffer=None, **kwargs):
+            # If copy method called, prevent recursion from replacing _shm.
+            if not hasattr(self, "_shm"):
+                self._shm = shm
+                self._proxied_type = existing_type
+            else:
+                # _proxied_type only used in pickling.
+                assert hasattr(self, "_proxied_type")
+            try:
+                existing_type.__init__(self, *args, **kwargs)
+            except:
```

bare except clause should be avoided (use `except Exception:` instead)

---

Agreed! I have changed this in a097dbb6fd2fe46be2715818338b874ba350824e as part of PR-11816.